ADT7470.cpp

```diff
@@ -243,16 +243,16 @@ bool ADT7470::setPulsesPerRevolution(uint8_t idx, uint8_t val)
   uint8_t reg;
   _read(ADT7470_FAN_PPR_REGISTER, &reg);
   reg &= ~mask;
-  reg |= (val << (idx * 2));
+  reg |= ((val - 1) << (idx * 2));
```

Comment:
OK, in the readme.md val is indeed defined as 1..4 so this  val -1 is correct.
TODO: Should be in the comments too in the ADT7470.h file 

TODO: val should be constrained to 1..4 before being used.